### PR TITLE
Run pre-commit with make

### PR DESCRIPTION
**Pull request by amolkahat**
Updating pre-commit hook with make. pre-commit-golang repo is no longer maintained[1]. Instead of using
unmaintained repo we can use make to run pre-commit.

[1] https://github.com/dnephin/pre-commit-golang/issues/98

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,15 +1,31 @@
 repos:
-- repo: https://github.com/dnephin/pre-commit-golang
-  rev: v0.5.1
-  hooks:
-    - id: go-fmt
-      exclude: ^vendor
-    - id: go-vet
-    - id: go-mod-tidy
-    - id: go-lint
```
Comment (Sandeepyadav93) [inline .pre-commit-config.yaml:9]: We are removing this but not adding `golint` back.

**Review by Sandeepyadav93 — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (amolkahat) [reply .pre-commit-config.yaml:9]: Done. Added golint.

**Comment by amolkahat** [comment]
Results
```
akahat@jatayu mariadb-operator $ pre-commit run -a
golangci-lint............................................................Passed
golint...................................................................Passed
gofmt....................................................................Passed
govet....................................................................Passed
gotidy...................................................................Passed
make-manifests...........................................................Passed
make-generate............................................................Passed
make-operator-lint.......................................................Passed
check for added large files..............................................Passed
fix utf-8 byte order marker..............................................Passed
check for case conflicts.................................................Passed
check that executables have shebangs.....................................Passed
check that scripts with shebangs are executable..........................Passed
check for merge conflicts................................................Passed
check for broken symlinks................................................Passed
detect destroyed symlinks................................................Passed
check yaml...............................................................Passed
check json...............................................................Passed
detect private key.......................................................Passed
fix end of files.........................................................Passed
don't commit to branch...................................................Passed
trim trailing whitespace.................................................Passed
bashate..................................................................Passed

akahat@jatayu mariadb-operator $
```

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**